incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -351,6 +351,10 @@ func (r *ObjectReconciler) isExcluded(log logr.Logger, inst *unstructured.Unstru
 		}
 		return false
 
+	// Object with nonempty finalizer list is not propagated
+	case inst != nil && len(inst.GetFinalizers()) != 0:
```

Comment:
Two things:

1. This code will never be called on Secrets - a Secret with finalizers will `return false` on line 352.
2. The rest of the code in this function assumes that `inst != nil` so you can probably assume that here as well.

To solve issue #1, I'd recommend moving this logic _above_ the switch statement, since this issue applies to all types.